Remove Block context manager

VincentLanglet opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 3.x is for everything backwards compatible, like patches, features and deprecation notices
    - 4.x is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataPageBundle/blob/3.x/CONTRIBUTING.md#base-branch
-->
I am targeting this branch, because {reason}.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

Closes #{put_issue_number_here}.

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
    This will end up on https://github.com/sonata-project/SonataPageBundle/releases,
    please keep it short and clear and to the point
-->

<!--
    If you are updating something that doesn't require
    a release, you can delete the whole "Changelog" section.
    (eg. update to docs, tests)
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Removed
- Remove `BlockContextManager` service to use the default one from BlockBundle.
```

<!--
    If this is a work in progress, uncomment the "To do" section.
    You can add as many tasks as you want.
    If some are not relevant, just remove them.
-->
<!--
## To do

- [ ] Update the tests;
- [ ] Update the documentation;
- [ ] Add an upgrade note.
-->

VincentLanglet committed:
Remove BlockContextManager

jordisala1991 commented on docs/reference/installation.rst:
```diff
@@ -138,16 +138,7 @@ SonataAdminBundle Configuration
-.. note::
-
-    Please you need to use the context ``sonata_page_bundle`` in the SonataBlockBundle to add block into a Page.
+You need to use the context ``sonata_page_bundle`` in the SonataBlockBundle to add block into a Page.
```
Do you know what does this means?

VincentLanglet replied:
No.

I saw that in tests we have
```
sonata_block:
    http_cache: false
    blocks:
        sonata.admin.block.admin_list:
            contexts: [admin]
    default_contexts: [sonata_page_bundle]
```
But i can't say it's related ; so I didn't try to change the doc

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

eerison commented:
Hey @VincentLanglet do you need that I test it? But what exactly do you need that I test :D

VincentLanglet commented:
In https://github.com/sonata-project/SonataPageBundle/issues/1495 you reported that you got the error

> at http://localhost:8001/admin/dashboard is returning An exception has been thrown during the rendering of a template ("The options "manager", "page_id" do not exist. Defined options are: "attr", "extra_cache_keys", "groups", "template", "ttl", "use_cache".").

I was wondering if my PR fixed the issue (or if you get another error which would explain why this class is useful)

eerison commented:
> In #1495 you reported that you got the error
> 
> > at http://localhost:8001/admin/dashboard is returning An exception has been thrown during the rendering of a template ("The options "manager", "page_id" do not exist. Defined options are: "attr", "extra_cache_keys", "groups", "template", "ttl", "use_cache".").
> 
> I was wondering if my PR fixed the issue (or if you get another error which would explain why this class is useful)

ohh Ok I got it, I'll check ;)

eerison commented:
Hey @VincentLanglet when I tried to install your branch I got this error

```
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  
!!  In ContainerBuilder.php line 1030:
!!                                                                                 
!!    You have requested a non-existent service "sonata.page.block.context_manage  
!!    r".                                                                          
!!                                                                                 
!!  
!!  
Script @auto-scripts was called via post-update-cmd
```

VincentLanglet commented:
> Hey @VincentLanglet when I tried to install your branch I got this error
> 
> ```
> Executing script cache:clear [KO]
>  [KO]
> Script cache:clear returned with error code 1
> !!  
> !!  In ContainerBuilder.php line 1030:
> !!                                                                                 
> !!    You have requested a non-existent service "sonata.page.block.context_manage  
> !!    r".                                                                          
> !!                                                                                 
> !!  
> !!  
> Script @auto-scripts was called via post-update-cmd
> ```

Do you have the config
```
sonata_block:
    context_manager: sonata.page.block.context_manager
```
?

I removed it from the doc, because I removed the service, i think you have to use the block manager from Block bundle so
`sonata.block.context_manager.default` (or just remove the line since it's the default value).

eerison commented:
> > Hey @VincentLanglet when I tried to install your branch I got this error
> > ```
> > Executing script cache:clear [KO]
> >  [KO]
> > Script cache:clear returned with error code 1
> > !!  
> > !!  In ContainerBuilder.php line 1030:
> > !!                                                                                 
> > !!    You have requested a non-existent service "sonata.page.block.context_manage  
> > !!    r".                                                                          
> > !!                                                                                 
> > !!  
> > !!  
> > Script @auto-scripts was called via post-update-cmd
> > ```
> 
> Do you have the config
> 
> ```
> sonata_block:
>     context_manager: sonata.page.block.context_manager
> ```
> 
> ?
> 
> I removed it from the doc, because I removed the service, i think you have to use the block manager from Block bundle so `sonata.block.context_manager.default` (or just remove the line since it's the default value).

Hey @VincentLanglet I removed this line `context_manager: sonata.page.block.context_manager` from my `sonata_page.yaml`, and I didn't get any issue with composer update

and my admin page is working as expect

<img width="1013" alt="Screenshot 2022-08-08 at 18 16 48" src="https://user-images.githubusercontent.com/6358755/183464782-f3407098-e7bf-4623-8db0-e16c52e31f39.png">


I don't know why were we using those `"manager", "page_id"`, but it looks fine 👀

eerison commented:
Hey @VincentLanglet should we add this note about

context_manager: sonata.page.block.context_manager? In upgrade notes? Or it's not necessary?

VincentLanglet commented:
> Hey @VincentLanglet should we add this note about
> 
> context_manager: sonata.page.block.context_manager? In upgrade notes? Or it's not necessary?

Indeed. We should even deprecate the context_manager in the 3.x branch (and maybe we dont even need it on 3.x).

So a PR on 3.x is needed. Do you want to do it or I do it ?

eerison commented:
> > Hey @VincentLanglet should we add this note about
> > context_manager: sonata.page.block.context_manager? In upgrade notes? Or it's not necessary?
> 
> Indeed. We should even deprecate the context_manager in the 3.x branch (and maybe we dont even need it on 3.x).
> 
> So a PR on 3.x is needed. Do you want to do it or I do it ?

Hey @VincentLanglet well for the free time that I'm getting I'm investing in this template issue :/

Then i can't do that  for now :/

eerison commented:
~@VincentLanglet FYI: when I try to use 3.x code without `context_manager: sonata.page.block.context_manager`~

~I got this error~

```
 [KO]
Script cache:clear returned with error code 255
!!  
!!  Fatal error: Class App\Entity\SonataPagePage contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Sonata\PageBundle\Model\PageInterface::getId) in /app/src/Entity/SonataPagePage.php on line 16
```

Sorry It is related with other thing!

eerison commented:
Hey @VincentLanglet  could you wait the Serializer PR pass? because after that I can test a page with blocks, I guess this functionality is used when the page has blocks! and for now it's not possible to test because it's broken :/ 

Because in 3.x if I remove this option (`context_manager: sonata.page.block.context_manager`)

I get this error
<img width="1362" alt="Screenshot 2022-08-09 at 08 37 21" src="https://user-images.githubusercontent.com/6358755/183581183-647b1ae1-d424-4e94-a8c9-5d2e9e14df4e.png">

VincentLanglet commented:
> Hey @VincentLanglet could you wait the Serializer PR pass? because after that I can test a page with blocks, I guess this functionality is used when the page has blocks! and for now it's not possible to test because it's broken :/
> 
> Because in 3.x if I remove this option (`context_manager: sonata.page.block.context_manager`)
> 
> I get this error <img alt="Screenshot 2022-08-09 at 08 37 21" width="1362" src="https://user-images.githubusercontent.com/6358755/183581183-647b1ae1-d424-4e94-a8c9-5d2e9e14df4e.png">

It could be interesting to know more about this error ; which class is complaining about the manager/page_id option ?

Looking at https://github.com/sonata-project/SonataPageBundle/search?p=1&q=page_id, I wonder if it's not cache related ; and since we remove this feature...

eerison commented:
> > Hey @VincentLanglet could you wait the Serializer PR pass? because after that I can test a page with blocks, I guess this functionality is used when the page has blocks! and for now it's not possible to test because it's broken :/
> > Because in 3.x if I remove this option (`context_manager: sonata.page.block.context_manager`)
> > I get this error <img alt="Screenshot 2022-08-09 at 08 37 21" width="1362" src="https://user-images.githubusercontent.com/6358755/183581183-647b1ae1-d424-4e94-a8c9-5d2e9e14df4e.png">
> 
> It could be interesting to know more about this error ; which class is complaining about the manager/page_id option ?
> 
> Looking at https://github.com/sonata-project/SonataPageBundle/search?p=1&q=page_id, I wonder if it's not cache related ; and since we remove this feature...

I'll try to clean up my cache, I'll post here the results :)

VincentLanglet commented:
> I'll try to clean up my cache, I'll post here the results :)

I meant it's maybe related to class inside the src/Cache folder.
Since we removed all this classes in 4.x it may explain why we don't need this class anymore.

Anyway, I'll wait for the other PR to be merged first

eerison commented:
> > I'll try to clean up my cache, I'll post here the results :)
> 
> I meant it's maybe related to class inside the src/Cache folder. Since we removed all this classes in 4.x it may explain why we don't need this class anymore.
> 
> Anyway, I'll wait for the other PR to be merged first

ohhh I got it!

yeah But I can't test my contact page yet, because of this issue #1496

SonataCI commented:
Could you please rebase your PR and fix merge conflicts?

VincentLanglet commented:
> > > I'll try to clean up my cache, I'll post here the results :)
> > 
> > 
> > I meant it's maybe related to class inside the src/Cache folder. Since we removed all this classes in 4.x it may explain why we don't need this class anymore.
> > Anyway, I'll wait for the other PR to be merged first
> 
> ohhh I got it!
> 
> yeah But I can't test my contact page yet, because of this issue #1496

Pr is rebased if you want to try again @eerison

eerison commented:
<img width="925" alt="Screenshot 2022-08-10 at 09 55 03" src="https://user-images.githubusercontent.com/6358755/183846916-2269fbdc-ecd0-44e1-bcae-bc91eb374958.png">
It works @VincentLanglet 🥳

VincentLanglet commented:
WDYT @jordisala1991 Should we go with this strategy ?

 If so, I am not sure it really useful to introduce some deprecation  in 3.x. if the option (context_manager: sonata.page.block.context_manager) cannot be removed without breaking the app.

jordisala1991 commented:
I its broken, IMO lets fix on 4.x by removing it, and lets see if someone complains. It seems to work and the branch is not released, so its not a big risk, we can recover that file later if needed. And I dont think we should do a deprecation here.

eerison commented:
> WDYT @jordisala1991 Should we go with this strategy ?
> 
> If so, I am not sure it really useful to introduce some deprecation in 3.x. if the option (context_manager: sonata.page.block.context_manager) cannot be removed without breaking the app.

I guess we could go with this, and make the 4.x branch usable, and when we going to do a final test we can try to cover this functionality, I can add this in the 4.0 tickets list!

eerison commented:
> I its broken, IMO lets fix on 4.x by removing it, and lets see if someone complains. It seems to work and the branch is not released, so its not a big risk, we can recover that file later if needed. And I dont think we should do a deprecation here.

exactly

VincentLanglet commented:
> I its broken, IMO lets fix on 4.x by removing it, and lets see if someone complains. It seems to work and the branch is not released, so its not a big risk, we can recover that file later if needed. And I dont think we should do a deprecation here.

Ready to review then @jordisala1991

jordisala1991 reviewed (dismissed):
OK for the code, IMO we could add some entry in the UPGRADE guide. to mention this was removed.

VincentLanglet committed:
Add Upgrade

VincentLanglet commented:
> OK for the code, IMO we could add some entry in the UPGRADE guide. to mention this was removed.

Added
